### PR TITLE
Use the first good image for the open graph tag

### PR DESCRIPTION
**Pull request by aforcier**
#### Changes proposed in this Pull Request:
This is a tweak to the open graph image selection logic, making it select the first 'good' image in a post instead of the last (the Twitter card tag logic already does this).

The image tags can be inspected by viewing the source of a post page, they look like this:

```html
<!-- Jetpack Open Graph Tags -->
<meta property="og:type" content="article" />
<meta property="og:title" content="A story post" />
<meta property="og:url" content="https://example.com/2020/10/20/a-story-post/" />
<meta property="og:description" content="Visit the post for more." />
<meta property="article:published_time" content="2020-10-20T06:39:37+00:00" />
<meta property="article:modified_time" content="2020-10-20T06:39:37+00:00" />
<meta property="og:site_name" content="alexforcier" />
<meta property="og:image" content="https://i2.wp.com/example.com/wp-content/uploads/2020/10/cropped-WP-Icon.png?fit=512%2C512&amp;ssl=1" />
<meta property="og:image:width" content="512" />
<meta property="og:image:height" content="512" />
<meta property="og:locale" content="en_US" />
<meta name="twitter:text:title" content="A story post" />
<meta name="twitter:image" content="https://i2.wp.com/example.com/wp-content/uploads/2020/10/cropped-WP-Icon.png?fit=240%2C240&amp;ssl=1" />
<meta name="twitter:card" content="summary" />
<meta name="twitter:description" content="Visit the post for more." />
```

This was originally done in https://github.com/Automattic/jetpack/pull/17198 and has been extracted here to its own PR.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Run the new unit test.

For any post with multiple images, inspect the image tags. The `og:image` tag should contain the first image in the post, or if the post contains a story block, the first slide in the story.

(The `twitter:image` tag behavior is unaffected by this PR, but you can observe it now has the same image as the `og:image` tag, where before they differed for posts with multiple images.)

#### Proposed changelog entry for your changes:

* Open Graph Tags: Use the first suitable image found in the post.

### CODE REVIEW

**Comment by aforcier** [comment]
@jeherve updated with a unit test as requested. The existing unit tests didn't test `jetpack_og_get_image` for individual posts, so I looked for ways to force `is_singular()` to be `true` for the test (so that [this conditional](https://github.com/Automattic/jetpack/blob/ad1c1e2ab5e1f535cf8ac0439a3a1bfbb0ff4cb0/functions.opengraph.php#L311) is `true`).

I found some similar handling in [test_class.jetpack-geo-location.php](https://github.com/Automattic/jetpack/blob/7e1597014c1b79a20b24b73386dbd6726e24ef69/tests/php/modules/geo-location/test_class.jetpack-geo-location.php) which I used as a reference, but I'm not at all well versed in WordPress unit testing. Please let me know if there's a better way to achieve this.

**Comment by jeherve** [comment]
> updated with a unit test as requested

Thank you!

I pushed a commit to simplify things a bit. This looks good to me as it is.

**Review by jeherve — APPROVED** [review]

**Review by fgiannar — DISMISSED** [review]

File: tests/php/test_class.functions.opengraph.php [hunk 1]
```diff
@@ -21,7 +22,12 @@ public function setUp() {
 	 * Include Open Graph functions after each test.
 	 */
 	public function tearDown() {
+		// Restoring global variables.
+		global $wp_the_query;
+		$wp_the_query = new WP_Query(); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
 		parent::tearDown();
```
[the remark this answers is not in the document]
Comment (jeherve) [reply tests/php/test_class.functions.opengraph.php:29]: 👍 Done:
821c93d17c0c81d52958cbcb1c94b8d5566d5a08

**Review by jeherve — APPROVED** [review]

**Comment by aforcier** [comment]
Thanks @jeherve and @fgiannar !

> I did move the milestone to 9.2 though, seeing as we're past the code freeze and I think we'd benefit from having this tested properly during the next Beta period. Since Open Graph impacts the look of posts on Facebook et al., every little change can sometimes result in quite a bit of support requests.

That sounds good to me! I updated one more `@since` tag to point to 9.2.

Just checking, since the branch for 9.1 has already been cut I'm okay to merge this in and deploy the WPcom change as usual right? Or should I wait until after 9.1's release on Nov 10?

**Comment by jeherve** [comment]
>  since the branch for 9.1 has already been cut I'm okay to merge this in and deploy the WPcom change as usual right? 

Yep, you can go ahead and merge!

**Review by jeherve — APPROVED** [review]

**Pull request merged**